docs/ref/models/indexes.txt

```diff
@@ -32,22 +32,63 @@ options`_.
 
 .. attribute:: Index.fields
 
-A list of the name of the fields on which the index is desired.
+A list of the name of the fields or database expressions on which the index is
+desired.
 
 By default, indexes are created with an ascending order for each column. To
 define an index with a descending order for a column, add a hyphen before the
-field's name.
+field's name. When using an expression, use ``.desc()`` for descending order.
```

Comment:
Maybe `nulls_first` and `nulls_last` for `asc` and `desc` could also be discussed in this paragraph, or somewhere else with examples?